components/cdc/src/observer.rs

```diff
@@ -130,32 +130,34 @@ impl<E: KvEngine> CmdObserver<E> for CdcObserver {
                 old_value_cache.access_count += 1;
                 if let Some((old_value, mutation_type)) = old_value_cache.cache.remove(&key) {
                     return match mutation_type {
-                        MutationType::Insert => (None, None),
-                        MutationType::Put | MutationType::Delete => match old_value {
-                            OldValue::None => (None, None),
-                            OldValue::Value {
-                                start_ts,
-                                short_value,
-                            } => {
-                                let mut statistics = None;
-                                let value = short_value.or_else(|| {
-                                    statistics = Some(Statistics::default());
+                        // Old value of an Insert is guaranteed to be None.
```

Comment:
Sometimes it is not true, the type of mutation is decided by TiDB, see https://github.com/tikv/tikv/pull/9872